pallets/gear/src/internal.rs

```diff
@@ -307,7 +307,11 @@ where
 
         // Validating duration.
         if hold.expected_duration().is_zero() {
-            unreachable!("Failed to figure out correct wait hold bound");
+            // TODO: Replace with unreachable call after:
+            // - `HoldBound` safety usage stabilized;
+            // - Issue #1173 solved.
+            log::error!("Failed to figure out correct wait hold bound");
+            return;
```

Comment:
do you plan to remove that asap? it's inappropriate to hold this solution for long, because you can loose messages in a such way.

Reply:
discussed